Allow bootstrap update on git worktree checkouts

ndevenish opened the pull request:
These have a `.git` in the root repository folder that is a file pointing to the real checkout. Rather than try and work out what to do, just assume that the user knows what they are doing in this scenario.

Anthchirp reviewed (approved):
Sensible, although the message is a bit misleading then

ndevenish commented:
Okay, so I fixed the message, with the assumption that if you have a `.git` file then that's what is going on - if you have a non-worktree git checkout where the `.git` file exists then I guess you'll have to live with the confusion 🤷 

I've also fixed precommitbx which also fell over on the same situation.
